index.bs

```diff
@@ -188,8 +188,11 @@ The requestStorageAccessForOrigin API defines a [=powerful feature=] identified
   <dd>
     <div algorithm='top-level-storage-access-key-generation'>
     To generate a new [=permission key=] for the "<a permission><code>top-level-storage-access</code></a>" feature, given an [=environment settings object=] |settings|, run the following steps:
+    1. Let |current origin| be |settings|' [=environment settings object/origin=].
+    1. If |current origin| is not [=same site=] with |settings|' [=top-level origin=], return null.
```

Comment:
small note: it wasn't super clear to me [in the spec](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-opaque) how to create an opaque origin, and it sounds like comparisons are possible, and it wasn't clear to me we wanted that vs just nulling this out.

Reply:
I think the main problem here is that you haven't specified a key comparison algorithm, which is mandatory if you generate a custom key, see https://www.w3.org/TR/permissions/#dfn-permission-key-generation-algorithm

You could specify that and then include a check for null, to be explicit?

Reply:
aha, yep, sorry about that. Added.